src/main/diagnostics/rdBinInShell.ts

```diff
@@ -0,0 +1,72 @@
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+
+import { PathManagementStrategy } from '@/integrations/pathManager';
+import mainEvents from '@/main/mainEvents';
+import { spawnFile } from '@/utils/childProcess';
+import paths from '@/utils/paths';
+
+import type { DiagnosticsCategory, DiagnosticsChecker } from './diagnostics';
+
+let pathStrategy = PathManagementStrategy.NotSet;
+
+mainEvents.on('settings-update', (cfg) => {
+  pathStrategy = cfg.pathManagementStrategy;
+});
+
+class RDBinInShell implements DiagnosticsChecker {
```

Comment:
Maybe a name like `RDBinInPath` or `RDBinInPATH` would be clearer?

Reply:
I've opted to go for `RDBinInShellPath` (because we don't actually test if it's in the system-level `PATH`).
Fixed in c344f47d400c7690d71d398c574302a3a084359e.